jetty-websocket/javax-websocket-client/src/main/java/org/eclipse/jetty/websocket/javax/client/JavaxWebSocketClientContainer.java

```diff
@@ -98,11 +95,18 @@ protected WebSocketExtensionRegistry getExtensionRegistry()
 
     protected HttpClient getHttpClient()
     {
-        return coreClient.getHttpClient();
+        return getWebSocketCoreClient().getHttpClient();
     }
 
-    protected WebSocketCoreClient getWebSocketCoreClient() throws Exception
+    protected WebSocketCoreClient getWebSocketCoreClient()
     {
+        if (coreClient == null)
+        {
+            coreClient = coreClientFactory.get();
```

Comment:
What if it's null?
What if it's not started?

Reply:
then we throw NPE and are broken.  Not really sure I want to create a default factory to handle the case that a passed in factory doesn't do it's job.   I could throw a ISE, but I don't think that is much better.